src/docfx/build/xref/XrefResolver.cs

```diff
@@ -137,21 +139,17 @@ public XrefMapModel ToXrefMapModel(bool isLocalizedBuild)
             var repositoryBranch = _repository?.Branch;
             var basePath = _config.BasePath.ValueWithLeadingSlash;
 
+            // DHS appends branch information from cookie cache to URL, which is wrong for UID resolved URL
+            // output xref map with URL appending "?branch=master" for master branch
             var references =
                 isLocalizedBuild
                 ? Array.Empty<ExternalXrefSpec>()
-                : _internalXrefMap.Value.Values
-                .Select(xref =>
-                {
-                    // DHS appends branch information from cookie cache to URL, which is wrong for UID resolved URL
-                    // output xref map with URL appending "?branch=master" for master branch
-                    var query = repositoryBranch != "live" ? $"?branch={repositoryBranch}" : "";
-                    var href = UrlUtility.MergeUrl($"https://{_xrefHostName}{xref.Href}", query);
-
-                    var xrefSpec = xref.ToExternalXrefSpec(href);
-                    return xrefSpec;
-                })
-                .OrderBy(xref => xref.Uid).ToArray();
+                : (from xrefs in _internalXrefMap.Value.Values
```

Comment:
I feel the original expression is more readable...
Just simply change to `xrefList` and `xrefList.First().ToExternalXrefSpec`